mage_ai/data_preparation/git/__init__.py

```diff
@@ -133,44 +134,50 @@ def wrapper(self, *args, **kwargs):
         return wrapper
 
     @_remote_command
-    def reset(self, branch: str = None):
+    def reset(self, branch: str = None) -> None:
         self.origin.fetch()
         if branch is None:
             branch = self.current_branch
         self.repo.git.reset('--hard', f'{self.origin.name}/{branch}')
 
     @_remote_command
-    def push(self):
+    def push(self) -> None:
         self.repo.git.push(
             '--set-upstream',
             self.origin.name,
             self.current_branch
         )
 
     @_remote_command
-    def pull(self):
+    def pull(self) -> None:
         self.origin.pull(self.current_branch)
+        self.__pip_install()
 
     def status(self) -> str:
         return self.repo.git.status()
 
-    def commit(self, message):
+    def commit(self, message) -> None:
         if self.repo.index.diff(None) or self.repo.untracked_files:
             self.repo.git.add('.')
             self.repo.index.commit(message)
 
-    def change_branch(self, branch):
+    def change_branch(self, branch) -> None:
         if branch in self.repo.heads:
             current = self.repo.heads[branch]
             current.checkout()
         else:
             current = self.repo.create_head(branch)
             current.checkout()
 
-    def __set_git_config(self):
+    def __set_git_config(self) -> None:
         if self.git_config.username:
             self.repo.config_writer().set_value(
                 'user', 'name', self.git_config.username).release()
         if self.git_config.email:
             self.repo.config_writer().set_value(
                 'user', 'email', self.git_config.email).release()
+
+    def __pip_install(self) -> None:
+        cmd = f'pip3 install -r {self.repo.working_dir}/requirements.txt'
```

Comment:
Let's make sure we only install the requirements.txt if it exists

Reply:
> Let's make sure we only install the requirements.txt if it exists

Good catch: made the suggested change.